composition-js/src/merging/merge.ts

```diff
@@ -967,7 +967,7 @@ class Merger {
     }
     for (const source of sources) {
       // As soon as we find a subgraph that has the type but not the field, we hint.
-      if (source && !source.field(field.name)) {
+      if (source && !source.field(field.name) && !this.typeIsFullyExternal(source)) {
```

Comment:
To provide the source index here, you'll want to change the above `for` loop into:
```
for (const [i, source] of sources.entries()) {
  // ...
}
```